File: 70.climbing-stairs.java
/*
 * @lc app=leetcode id=70 lang=java
 *
 * [70] Climbing Stairs
 */

// @lc code=start

import java.util.ArrayList;
import java.util.List;

class Solution {
    public int climbStairs(int n) {
        List<Integer> dp = new ArrayList<>();
        dp.add(-1);
        dp.add(1);
        dp.add(2);
        for (int ix = 3; ix <= n; ++ix) {
            dp.add(dp.get(ix - 1) + dp.get(ix - 2));
        }
        return dp.get(n);
    }

}
// @lc code=end

File: 70.climbing-stairs.cpp
/*
 * @lc app=leetcode id=70 lang=cpp
 *
 * [70] Climbing Stairs
 */

// @lc code=start

#include <vector>

using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        vector<int> dp(max(n + 1, 3), 0);
        dp[0] = -1; dp[1] = 1; dp[2] = 2;
        if (n < 3) return dp[n];
        for (int ix = 3; ix <= n; ++ix) {
            dp[ix] = dp[ix - 1] + dp[ix - 2];
        }
        return dp[n];
    }
};
// @lc code=end
